### PR TITLE
fixed rendering for all rotations, unit tests added

### PR DESCRIPTION
**Pull request by mkroutikov**

### CODE REVIEW

**Review by YinlinHu — COMMENTED** [review]
After playing with the raw PDFium API, I found the rotation flag is unnecessary, the XY offsets in FPDF_RenderPageBitmapWithMatrix have already taken the rotation into account. It seems you do the transformation in redstork, and again you do the reverse transformation in redstork-ui, which I think is unnecessary. Please see a working clip version based on the raw API (tested under four rotation cases):
https://github.com/YinlinHu/kuafu/blob/a9f1155eba3311e10a8a3e84f889581376ad26c8/kuafu/pdfworker.py#L312

**Comment by mkroutikov** [comment]
This is by design.

If you ever display only full page (cropbox), then FPDF api is sufficient.

If you want to use PDF coordinates to render page objects, then rotation has to be undone.

Redstork needs this ability to render individual pieces of a page, and it is much more convenient if API just takes the native PDF coordinates (because these coordinates are reported by API when objects are measured).

FPDF renderer does the following: (1) flip coordinate system (from bottom-to-top PDF to top-to-bottom image), then applies the rotation and shifts the page to get the final output. 

Redstork reverses this, so that `rotation` flag is not taken into account, and rendering can be done directly in PDF coordinates.
For example, I can render a page using its MediaBox or CropBox (sometimes they differ).

Basically, redstork treats `rotation` value as a "user convenience" feature, and its removed from the rendering API. Yes, sample UI does automatic rotation, much like AcroReader of FDF.

Hope this makes sense.

**Comment by YinlinHu** [comment]
It makes perfect sense. Thanks for your clarification.
